### PR TITLE
Modified te save_page endpoint

### PR DESCRIPTION
**Pull request by AndrewCzh**
Modified the save_page endpoint in the API to allow users to pass 'text' and 'url' parameters. The endpoint now fetches the HTML content of the provided URL, along with the text, stores in the page_data_store.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -4,4 +4,7 @@ click==8.0.4
 ruff
 httpx==0.25.2
 fastapi>=0.68.0
-uvicorn[standard]>=0.14.0
\ No newline at end of file
+uvicorn[standard]>=0.14.0
+requests~=2.31.0
+pydantic~=2.5.2
```
Comment (Olivernyu) [inline requirements.txt]: Is it really necessary to add `pydantic` as a dependency? It is already a dependency of fastapi that gets installed when one installs fastapi.

File: .DS_Store
Comment (Olivernyu) [inline .DS_Store:1]: Can you remove this file from the commit?

**Review by Olivernyu — CHANGES_REQUESTED** [review]
The functional changes look good to me. Can you check the comments i left on `requirements.txt`, there may be some redundant dependencies. Also, could you remove `.DS_Store` from the commits. I assume it got added by mistake.

File: requirements.txt [hunk 2]
```diff
@@ -8,4 +8,5 @@ uvicorn[standard]>=0.14.0
 requests~=2.31.0
 pydantic~=2.5.2
 beautifulsoup4~=4.12.2
-transformers~=4.36.2
\ No newline at end of file
+transformers~=4.36.2
+torch~=2.1.2
```
Comment (Alex-YuYan) [inline requirements.txt]: From our design doc, we don't need to run any model locally, so we probably don't need torch.

File: app/routers/summarize_endpoint.py [hunk 3]
```diff
@@ -1,17 +1,32 @@
...
+from transformers import pipeline
 from ..shared_resources import page_data_store
 
 router = APIRouter()
 
+summarizer = pipeline("summarization")
+
+
+def summarize_text(text):
+    summary = summarizer(text, max_length=500, min_length=30, do_sample=False)
```
Comment (Alex-YuYan) [inline app/routers/summarize_endpoint.py]: According to design doc, we will be using OpenAI API for the summarization. It is ok to have it here for now, but we will probably update it in the future.

**Review by Alex-YuYan — CHANGES_REQUESTED** [review]

File: app/routers/summarize_endpoint.py (hunk 3 above)
Comment (AndrewCzh) [reply app/routers/summarize_endpoint.py]: For sure.

File: requirements.txt (hunk 1 above)
Comment (AndrewCzh) [reply requirements.txt]: Delete pydantic and requests.

**Review by Olivernyu — DISMISSED** [review]
Thanks for making the changes, LGTM!

**Review by Alex-YuYan — DISMISSED** [review]
LGTM. Let's change the summarizer later.

**Review by Alex-YuYan — DISMISSED** [review]

File: app/routers/save_endpoint.py [hunk 4]
```diff
@@ -21,9 +32,14 @@ async def some_func(page_id: str):
             page_data = page_data_store.get(page_id)
             ...
     """
+    text = page_data.text
+    url = page_data.url
+    html_content = fetch_html(url)
```
Comment (leo4life2) [inline app/routers/save_endpoint.py:37]: `fetch_html` could throw an error. consider wrapping this line in a `try ... except` block.

File: app/routers/summarize_endpoint.py [hunk 5]
```diff
@@ -1,17 +1,47 @@
 from fastapi import APIRouter, HTTPException
+from openai import OpenAI
+
+# import os
+
 from ..shared_resources import page_data_store
 
 router = APIRouter()
+# api_key = os.getenv("OPENAI_API_KEY")
```
Comment (leo4life2) [inline app/routers/summarize_endpoint.py:9]: I think we can keep it like this? Or do `os.getenv("OPENAI_API_KEY", "Sample api key")`

**Review by Alex-YuYan — APPROVED** [review]

**Review by leo4life2 — APPROVED** [review]

**Pull request merged**